privaterelay/tests/utils_tests.py

```diff
@@ -20,3 +37,190 @@ def test_get_premium_country_lang(self):
+    flag = Flag.objects.get(name=TEST_FLAG_NAME)
+    assert waffle_cache.get(flag._cache_key(TEST_FLAG_NAME)) == flag
+
+
+def test_flag_is_active_for_task_missing_flag_created_with_default_true(
+    flag_user: User | None,
+    waffle_settings: SettingsWrapper,
+    waffle_cache: BaseCache,
+    caplog: LogCaptureFixture,
+) -> None:
+    waffle_settings.WAFFLE_CREATE_MISSING_FLAGS = True
+    waffle_settings.WAFFLE_FLAG_DEFAULT = True
+    assert flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+    assert caplog.record_tuples == []
+    flag = Flag.objects.get(name=TEST_FLAG_NAME)
+    assert waffle_cache.get(flag._cache_key(TEST_FLAG_NAME)) == flag
+
+
+def test_flag_is_active_for_task_missing_flag_with_default_true(
+    flag_user: User | None, waffle_settings: SettingsWrapper, caplog: LogCaptureFixture
+) -> None:
+    waffle_settings.WAFFLE_FLAG_DEFAULT = True
+    assert flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+    assert caplog.record_tuples == []
+    assert not Flag.objects.filter(name=TEST_FLAG_NAME).exists()
+
+
+def test_flag_is_active_for_task_existing_but_off_flag(flag_user: User | None) -> None:
+    flag = Flag.objects.create(name=TEST_FLAG_NAME)
+    assert flag.everyone is None
+    assert not flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+
+
+def test_flag_is_active_for_task_with_everyone_on_flag(flag_user: User | None) -> None:
+    Flag.objects.create(name=TEST_FLAG_NAME, everyone=True)
+    assert flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+
+
+def test_flag_is_active_for_task_with_everyone_off_flag(flag_user: User | None) -> None:
+    Flag.objects.create(name=TEST_FLAG_NAME, everyone=False)
+    assert not flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+
+
+def test_flag_is_active_for_task_user_flag(flag_user: User | None) -> None:
+    flag = Flag.objects.create(name=TEST_FLAG_NAME)
+    assert not flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+    if flag_user is None:
+        return  # Nothing further to test without a user
+    flag.users.add(flag_user)
+    assert flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+
+
+def test_flag_is_active_for_task_group_flag(flag_user: User | None) -> None:
+    flag = Flag.objects.create(name=TEST_FLAG_NAME)
+    assert not flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+    if flag_user is None:
+        return  # Nothing further to test without a user
+    group = Group.objects.create(name=TEST_FLAG_NAME)
+    flag.groups.add(group)
+    assert not flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+    group.user_set.add(flag_user)
+    assert flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+
+
+def test_flag_is_active_for_task_everyone_overrides_user(
+    flag_user: User | None,
+) -> None:
+    if flag_user is None:
+        return  # Nothing further to test without a user
+    flag = Flag.objects.create(name=TEST_FLAG_NAME, everyone=False)
+    flag.users.add(flag_user)
+    assert not flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
```

Comment:
@say-yawn this is the test that check everyone=False, user=True returns that the flag is inactive